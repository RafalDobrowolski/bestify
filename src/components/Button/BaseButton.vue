<template>
  <button
    class="b-btn"
    :class="buttonClass"
    v-bind="$attrs"
    v-on="$listeners"
  >
    <slot></slot>
  </button>
</template>

<script>
  import { TYPE } from "./BaseButton.types";

  export default {
    name: "bButon",
    props: {
      type: {
        type: String,
        default: ""
      }
   },
    computed: {
      buttonClass() {
        return {
          "b-btn__primary": this.type === TYPE.PRIMARY,
          "b-btn__secondary": this.type === TYPE.SECONDARY
        }
      }
    },
  }
</script>

<style lang="scss" scoped>
  @import "../../../src/scss/utils/_variables.scss";

  .b-btn {
    display: inline-flex;
    align-items: center;
    justify-content: center;
    border: 1px solid  #dbdbdb;
    border-radius: 4px;
    box-shadow: none;
    font-size: 1rem;
    font-weight: 400;
    cursor: pointer;
    padding: .5rem 1rem;
    line-height: 1.5;
    text-align: center;
    white-space: nowrap;
    background-color: #fff;
    color: #363636;

    &:hover {
      text-decoration: none;
    }
    &:focus {
      outline: none;
    }

    &__primary {
      background-color: $color-purple-200;
      color: $color-white;
      border-color: transparent;

      &:hover {
        background-color: $color-purple-400;
      }
    }

    &__secondary {
      background-color: $color-green-100;
      color: $color-white;
      border-color: transparent;

      &:hover {
        background-color: $color-green-200;
      }
    }
}
</style>