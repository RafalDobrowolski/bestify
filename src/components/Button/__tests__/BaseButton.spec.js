import { mount } from '@vue/test-utils';
import bButton from "../BaseButton";
import { TYPE } from "../BaseButton.types";

describe('Button', () => {
  test('Primary button is rendered', () => {
    const wrapper = mount(bButton, {
      propsData: {
          type: TYPE.PRIMARY
      }
    });
    expect(wrapper.classes('b-btn__primary')).toBe(true);
  });

  test('Button is rendered correctly', () => {
    const wrapper = mount(bButton);
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('Secondary button is rendered', () => {
    const wrapper = mount(bButton, {
      propsData: {
          type: TYPE.SECONDARY
      }
    });
    expect(wrapper.classes('b-btn__secondary')).toBe(true);
  });

  test('Button emit a click event', () => {
    const wrapper = mount(bButton);
    wrapper.trigger('click');
    const emmitedEvent = wrapper.emitted('click');
    expect(emmitedEvent).toHaveLength(1);
  });
});