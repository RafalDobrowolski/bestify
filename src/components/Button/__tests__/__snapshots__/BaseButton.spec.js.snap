// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Button is rendered correctly 1`] = `<button class="b-btn"></button>`;
